package.json

```diff
@@ -84,27 +85,203 @@
+    "icons": {
+      "camel-run": {
+        "description": "Run Camel Application",
+        "default": {
+          "fontPath": "./icons/camel-run.woff",
+          "fontCharacter": "\\E900"
+        }
+      }
+    },
     "commands": [
       {
         "command": "kaoto.open",
-        "title": "Open with Kaoto Graphical Editor for Camel"
+        "title": "Open with Kaoto"
       },
       {
-        "command": "kaoto.open.textualeditor",
-        "title": "Open Camel file with textual editor on the side",
-        "shortTitle": "Show source",
+        "command": "kaoto.open.source",
+        "title": "Open Source Code",
+        "shortTitle": "Show Source",
         "enablement": "activeCustomEditorId == webviewEditorsKaoto",
-        "category": "Camel",
-        "icon": "$(go-to-file)"
+        "category": "Kaoto",
+        "icon": "$(code)"
+      },
+      {
+        "command": "kaoto.close.source",
+        "title": "Close Source Code",
+        "shortTitle": "Close Source",
+        "enablement": "resourceFilename =~ /\\.(yaml)$/",
+        "category": "Kaoto",
+        "icon": "$(code)"
+      },
+      {
+        "command": "kaoto.new.file",
+        "title": "Camel File",
+        "category": "Kaoto",
+        "enablement": "false",
+        "icon": "$(new-file)"
+      },
+      {
+        "command": "kaoto.camel.jbang.export",
+        "title": "Export",
+        "category": "Kaoto",
+        "icon": "$(link-external)"
+      },
+      {
+        "command": "kaoto.camel.jbang.routes.yaml",
+        "shortTitle": "Create a Route",
+        "title": "Create a Camel Route using YAML DSL",
+        "category": "Kaoto",
+        "enablement": "workspaceFolderCount != 0"
+      },
+      {
+        "command": "kaoto.camel.jbang.routes.kamelet.yaml",
+        "shortTitle": "Create a Kamelet",
+        "title": "Create a Kamelet using YAML DSL",
+        "category": "Kaoto",
+        "enablement": "workspaceFolderCount != 0"
+      },
+      {
+        "command": "kaoto.camel.jbang.routes.pipe.yaml",
+        "shortTitle": "Create a Pipe",
+        "title": "Create a Custom Resource Pipe using YAML DSL",
+        "category": "Kaoto",
+        "enablement": "workspaceFolderCount != 0"
+      },
+      {
+        "command": "kaoto.camel.jbang.project.quarkus.new",
+        "title": "Create a Camel Quarkus project",
+        "category": "Kaoto",
+        "enablement": "workspaceFolderCount != 0"
+      },
+      {
+        "command": "kaoto.camel.jbang.project.springboot.new",
+        "title": "Create a Camel on SpringBoot project",
+        "category": "Kaoto",
+        "enablement": "workspaceFolderCount != 0"
+      },
+      {
+        "command": "kaoto.integrations.refresh",
+        "title": "Refresh",
+        "category": "Kaoto",
+        "icon": "$(refresh)"
+      },
+      {
+        "command": "kaoto.integrations.editSource",
+        "title": "Show Source",
+        "category": "Kaoto",
+        "icon": "$(code)"
+      },
```

Comment:
when called from command-palette ther eis an error:
![image](https://github.com/user-attachments/assets/1132f2fb-84aa-4002-8d23-14bd4caa3eaf)

Either hide it from command palette or implement it and filter on Kaoto editor

Reply:
in fact, i think that most of the commands are not implemented when called from command palette so need to hide them as first iteration.

![image](https://github.com/user-attachments/assets/99d3384c-7ec0-40ef-af8e-983267bc863a)
![image](https://github.com/user-attachments/assets/2e94dccf-6b64-40bc-8775-6604ba347e90)

Reply:
Yeah, they are not meant to be executed through the command palette. We should not add them there.

Reply:
good catch, we can disable all except maybe commands for views as Refresh eg.